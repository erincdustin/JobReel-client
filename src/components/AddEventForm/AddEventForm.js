import React from 'react';
import { Input, Label } from '../../components/Form/Form';
import Button from '../../components/Button/Button';
import JobReelContext from '../../context/JobReelContext';
import jobReelApiService from '../../services/jobreel-api-service';
import './AddEventForm.css';

class AddEventForm extends React.Component {

  static contextType = JobReelContext

  state = {
    error: null
  }

  handleSubmit = (e) => {
    e.preventDefault();
    e.persist();
    this.setState({ error: null }); 
    const event_name = e.target.event.value;
    const host = e.target.host.value;
    const city = e.target.city.value;
    const state = e.target.state.value;
    const address = e.target.address.value;
    const date = e.target.date.value;
    const url = e.target.url.value;
    const description = e.target.desc.value;
    const status = e.target.status.value;
    const userInput = { user_id: this.context.user.id, event_name, host, city, state, address, date, url, description, status };
    jobReelApiService.submitEvent(userInput)
      .then(res => {
        e.target.event.value = '';
        e.target.host.value = '';
        e.target.city.value = '';
        e.target.state.value = '';
        e.target.address.value = '';
        e.target.date.value = '';
        e.target.url.value = '';
        e.target.desc.value = '';
        e.target.status.value = '';
        this.context.setSavedEvents([...this.context.savedEvents, res]);
        this.context.setManualEventAdd(false)
      })
      .catch(res => {
        this.setState({ error: res.error })
      })
      
  }

  renderStateOptions = () => {
    const states = [
      ['Arizona', 'AZ'],
      ['Alabama', 'AL'],
      ['Alaska', 'AK'],
      ['Arkansas', 'AR'],
      ['California', 'CA'],
      ['Colorado', 'CO'],
      ['Connecticut', 'CT'],
      ['Delaware', 'DE'],
      ['Florida', 'FL'],
      ['Georgia', 'GA'],
      ['Hawaii', 'HI'],
      ['Idaho', 'ID'],
      ['Illinois', 'IL'],
      ['Indiana', 'IN'],
      ['Iowa', 'IA'],
      ['Kansas', 'KS'],
      ['Kentucky', 'KY'],
      ['Louisiana', 'LA'],
      ['Maine', 'ME'],
      ['Maryland', 'MD'],
      ['Massachusetts', 'MA'],
      ['Michigan', 'MI'],
      ['Minnesota', 'MN'],
      ['Mississippi', 'MS'],
      ['Missouri', 'MO'],
      ['Montana', 'MT'],
      ['Nebraska', 'NE'],
      ['Nevada', 'NV'],
      ['New Hampshire', 'NH'],
      ['New Jersey', 'NJ'],
      ['New Mexico', 'NM'],
      ['New York', 'NY'],
      ['North Carolina', 'NC'],
      ['North Dakota', 'ND'],
      ['Ohio', 'OH'],
      ['Oklahoma', 'OK'],
      ['Oregon', 'OR'],
      ['Pennsylvania', 'PA'],
      ['Rhode Island', 'RI'],
      ['South Carolina', 'SC'],
      ['South Dakota', 'SD'],
      ['Tennessee', 'TN'],
      ['Texas', 'TX'],
      ['Utah', 'UT'],
      ['Vermont', 'VT'],
      ['Virginia', 'VA'],
      ['Washington', 'WA'],
      ['West Virginia', 'WV'],
      ['Wisconsin', 'WI'],
      ['Wyoming', 'WY'],
    ];
    return states.map(state => {
      return <option key={state[1]} value={state[1]}>{state[0]}</option>
    })
  }

  render() {
    const { error } = this.state;
    return (
      <div className="AddEventForm">
        <div role='alert'>
          {error && <p>{error}</p>}
        </div>
        <form onSubmit={this.handleSubmit}>
          <div>
            <Label htmlFor='event-name-input'>
              Event
            </Label>
            <br/>
            <Input
              id='event-name-input'
              name='event'
            />
          </div>
          <div>
            <Label htmlFor='host-input'>
              Host
            </Label>
            <br/>
            <Input
              id='host-input'
              name='host'
            />
          </div>
          <div>
            <Label htmlFor='city-input'>
              City
            </Label>
            <br/>
            <Input
              id='city-input'
              name='city'
            />
          </div>
          <div>
            <Label htmlFor='state-input'>
              State
            </Label>
            <br/>
            <select name="state" id="state-input">
              {this.renderStateOptions()}
            </select>
          </div>
          <div>
            <Label htmlFor='address-input'>
              Address
            </Label>
            <br/>
            <Input
              id='address-input'
              name='address'
            />
          </div>
          <div>
            <Label htmlFor='date-input'>
              Date
            </Label>
            <br/>
            <Input
              type="date"
              id='date-input'
              name='date'
            />
          </div>
          <div>
            <Label htmlFor='url-input'>
              URL
            </Label>
            <br/>
            <Input
              id='url-input'
              name='url'
            />
          </div>
          <div>
            <Label htmlFor='desc-input'>
              Description
            </Label>
            <br/>
            <Input
              id='desc-input'
              name='desc'
            />
          </div>
          <div>
            <Label htmlFor='status-input'>
              Status
            </Label>
            <br/>
            <select
              id='status-input'
              name='status'
            >
              <option value="Will Attend">Will attend</option>
              <option value="Maybe">Maybe</option>
              <option value="Attended">Attended</option>
              <option value="Did Not Attend">Did Not Attend</option>
            </select>
          </div>
          <Button onClick={() => this.context.setManualEventAdd(false)} type="button">Back</Button>
          <Button type="submit">Submit</Button>
        </form>
      </div>
    )
  }
}

export default AddEventForm;