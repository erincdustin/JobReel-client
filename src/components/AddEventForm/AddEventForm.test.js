import React from 'react';
import ReactDOM from 'react-dom'
import AddEventForm from './AddEventForm'

describe('AddEventForm smoke test', () => {
  it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<AddEventForm />, div);
      ReactDOM.unmountComponentAtNode(div);
  })
})