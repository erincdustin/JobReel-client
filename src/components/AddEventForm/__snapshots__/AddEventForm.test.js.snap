// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddEventForm snapshot test renders the UI as expected 1`] = `
<div
  className="AddEventForm"
>
  <div
    role="alert"
  />
  <form
    onSubmit={[Function]}
  >
    <div>
      <Label
        htmlFor="event-name-input"
      >
        Event
      </Label>
      <br />
      <ForwardRef
        id="event-name-input"
        name="event"
      />
    </div>
    <div>
      <Label
        htmlFor="host-input"
      >
        Host
      </Label>
      <br />
      <ForwardRef
        id="host-input"
        name="host"
      />
    </div>
    <div>
      <Label
        htmlFor="city-input"
      >
        City
      </Label>
      <br />
      <ForwardRef
        id="city-input"
        name="city"
      />
    </div>
    <div>
      <Label
        htmlFor="state-input"
      >
        State
      </Label>
      <br />
      <select
        id="state-input"
        name="state"
      >
        <option
          key="AZ"
          value="AZ"
        >
          Arizona
        </option>
        <option
          key="AL"
          value="AL"
        >
          Alabama
        </option>
        <option
          key="AK"
          value="AK"
        >
          Alaska
        </option>
        <option
          key="AR"
          value="AR"
        >
          Arkansas
        </option>
        <option
          key="CA"
          value="CA"
        >
          California
        </option>
        <option
          key="CO"
          value="CO"
        >
          Colorado
        </option>
        <option
          key="CT"
          value="CT"
        >
          Connecticut
        </option>
        <option
          key="DE"
          value="DE"
        >
          Delaware
        </option>
        <option
          key="FL"
          value="FL"
        >
          Florida
        </option>
        <option
          key="GA"
          value="GA"
        >
          Georgia
        </option>
        <option
          key="HI"
          value="HI"
        >
          Hawaii
        </option>
        <option
          key="ID"
          value="ID"
        >
          Idaho
        </option>
        <option
          key="IL"
          value="IL"
        >
          Illinois
        </option>
        <option
          key="IN"
          value="IN"
        >
          Indiana
        </option>
        <option
          key="IA"
          value="IA"
        >
          Iowa
        </option>
        <option
          key="KS"
          value="KS"
        >
          Kansas
        </option>
        <option
          key="KY"
          value="KY"
        >
          Kentucky
        </option>
        <option
          key="LA"
          value="LA"
        >
          Louisiana
        </option>
        <option
          key="ME"
          value="ME"
        >
          Maine
        </option>
        <option
          key="MD"
          value="MD"
        >
          Maryland
        </option>
        <option
          key="MA"
          value="MA"
        >
          Massachusetts
        </option>
        <option
          key="MI"
          value="MI"
        >
          Michigan
        </option>
        <option
          key="MN"
          value="MN"
        >
          Minnesota
        </option>
        <option
          key="MS"
          value="MS"
        >
          Mississippi
        </option>
        <option
          key="MO"
          value="MO"
        >
          Missouri
        </option>
        <option
          key="MT"
          value="MT"
        >
          Montana
        </option>
        <option
          key="NE"
          value="NE"
        >
          Nebraska
        </option>
        <option
          key="NV"
          value="NV"
        >
          Nevada
        </option>
        <option
          key="NH"
          value="NH"
        >
          New Hampshire
        </option>
        <option
          key="NJ"
          value="NJ"
        >
          New Jersey
        </option>
        <option
          key="NM"
          value="NM"
        >
          New Mexico
        </option>
        <option
          key="NY"
          value="NY"
        >
          New York
        </option>
        <option
          key="NC"
          value="NC"
        >
          North Carolina
        </option>
        <option
          key="ND"
          value="ND"
        >
          North Dakota
        </option>
        <option
          key="OH"
          value="OH"
        >
          Ohio
        </option>
        <option
          key="OK"
          value="OK"
        >
          Oklahoma
        </option>
        <option
          key="OR"
          value="OR"
        >
          Oregon
        </option>
        <option
          key="PA"
          value="PA"
        >
          Pennsylvania
        </option>
        <option
          key="RI"
          value="RI"
        >
          Rhode Island
        </option>
        <option
          key="SC"
          value="SC"
        >
          South Carolina
        </option>
        <option
          key="SD"
          value="SD"
        >
          South Dakota
        </option>
        <option
          key="TN"
          value="TN"
        >
          Tennessee
        </option>
        <option
          key="TX"
          value="TX"
        >
          Texas
        </option>
        <option
          key="UT"
          value="UT"
        >
          Utah
        </option>
        <option
          key="VT"
          value="VT"
        >
          Vermont
        </option>
        <option
          key="VA"
          value="VA"
        >
          Virginia
        </option>
        <option
          key="WA"
          value="WA"
        >
          Washington
        </option>
        <option
          key="WV"
          value="WV"
        >
          West Virginia
        </option>
        <option
          key="WI"
          value="WI"
        >
          Wisconsin
        </option>
        <option
          key="WY"
          value="WY"
        >
          Wyoming
        </option>
      </select>
    </div>
    <div>
      <Label
        htmlFor="address-input"
      >
        Address
      </Label>
      <br />
      <ForwardRef
        id="address-input"
        name="address"
      />
    </div>
    <div>
      <Label
        htmlFor="date-input"
      >
        Date
      </Label>
      <br />
      <ForwardRef
        id="date-input"
        name="date"
        type="date"
      />
    </div>
    <div>
      <Label
        htmlFor="url-input"
      >
        URL
      </Label>
      <br />
      <ForwardRef
        id="url-input"
        name="url"
      />
    </div>
    <div>
      <Label
        htmlFor="desc-input"
      >
        Description
      </Label>
      <br />
      <ForwardRef
        id="desc-input"
        name="desc"
      />
    </div>
    <div>
      <Label
        htmlFor="status-input"
      >
        Status
      </Label>
      <br />
      <select
        id="status-input"
        name="status"
      >
        <option
          value="Will Attend"
        >
          Will attend
        </option>
        <option
          value="Maybe"
        >
          Maybe
        </option>
        <option
          value="Attended"
        >
          Attended
        </option>
        <option
          value="Did Not Attend"
        >
          Did Not Attend
        </option>
      </select>
    </div>
    <ForwardRef
      onClick={[Function]}
      type="button"
    >
      Back
    </ForwardRef>
    <ForwardRef
      type="submit"
    >
      Submit
    </ForwardRef>
  </form>
</div>
`;
